src/main/java/org/opensearch/flowframework/transport/DeprovisionWorkflowTransportAction.java

```diff
@@ -226,15 +226,23 @@ private void updateWorkflowState(
     ) {
         if (remainingResources.isEmpty()) {
             // Successful deprovision, reset state to initial
-            flowFrameworkIndicesHandler.putInitialStateToWorkflowState(
-                workflowId,
-                getUserContext(client),
-                ActionListener.wrap(indexResponse -> {
-                    logger.info("Reset workflow {} state to NOT_STARTED", workflowId);
-                }, exception -> { logger.error("Failed to reset to initial workflow state for {}", workflowId, exception); })
-            );
-            // return workflow ID
-            listener.onResponse(new WorkflowResponse(workflowId));
+            flowFrameworkIndicesHandler.doesTemplateExist(workflowId, templateExists -> {
+                if (Boolean.TRUE.equals(templateExists)) {
```

Comment:
nit: think this can be changed to `if (templateExists)`

Reply:
> nit: think this can be changed to `if (templateExists)`

It could, but I believe this is more correct (my linter complains otherwise!). 

`templateExists` is a `Boolean` but the `if(...)` syntax assumes a primitive `boolean`.  This results in unnecessary unboxing.  Furthermore, there's a possibility that it could be `null` and the unboxing would cause an NPE.

Reply:
sounds good